homeassistant/components/tradfri/coordinator.py

```diff
@@ -0,0 +1,102 @@
+"""Tradfri DataUpdateCoordinator."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from datetime import timedelta
+import logging
+from typing import Any
+
+from pytradfri.command import Command
+from pytradfri.device import Device
+from pytradfri.error import RequestError
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import SCAN_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TradfriDeviceDataUpdateCoordinator(DataUpdateCoordinator):
+    """Coordinator to manage data for a specific Tradfri device."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        *,
+        api: Callable[[Command | list[Command]], Any],
+        device: Device,
+    ) -> None:
+        """Initialize device coordinator."""
+        self.api = api
+        self.device = device
+        self._hub_available = True
+        self._exception: Exception | None = None
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=f"Update coordinator for {device}",
+            update_interval=timedelta(seconds=SCAN_INTERVAL),
+        )
+
+    @callback
+    def set_hub_available(self, available: bool) -> None:
+        """Set status of hub."""
+        if available != self._hub_available:
```

Comment:
We can just compare with `self.last_update_success`.
```suggestion
        if available != self.last_update_success:
```